biz.aQute.bndlib/src/aQute/bnd/service/reporter/WorkspaceEntryPlugin.java

```diff
@@ -0,0 +1,32 @@
+package aQute.bnd.service.reporter;
+
+import org.osgi.annotation.versioning.ProviderType;
+
+import aQute.bnd.build.Workspace;
+import aQute.bnd.osgi.Processor;
+
+/**
+ * This plugin extracts a piece of information from a BND workspace and converts
+ * it into a DTO representation. Its result will contributes to the formation of
+ * a workspace metadata report.
+ */
+@ProviderType
+public interface WorkspaceEntryPlugin {
+
+	/**
+	 * Extracts a piece of information from the BND workspace in arguments.
+	 * 
+	 * @param workspace the workspace to inspect, must not be {@code null}
+	 * @param reporter used to report error, must not be {@code null}
+	 * @return a DTO representation or {@code null} if no data is available
+	 */
+	public Object extract(final Workspace workspace, final Processor reporter) throws Exception;
```

Comment:
The use of Object as the return type in these plugins is a bit disconcerting. It means that only reflective access to the DTOs is possible. Why would we not want to define the DTO types and then return them here?

Reply:
Same as for BundleEntryPlugin.